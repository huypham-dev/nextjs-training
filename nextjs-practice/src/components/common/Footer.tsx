import { memo } from "react";
import Link from "next/link";

const Footer = (): JSX.Element => (
  <footer className="w-full bg-white px-6 border-t">
    <div className="container mx-auto max-w-4xl py-6 flex flex-wrap md:flex-no-wrap justify-between items-center text-sm">
      &copy;2021 Your Company. All rights reserved.
      <div className="pt-4 md:p-0 text-center md:text-right text-xs">
        <Link href="/policy">
          <a className="text-black no-underline hover:underline">
            Privacy Policy
          </a>
        </Link>
        <Link href="/condition">
          <a className="text-black no-underline hover:underline ml-4">
            Terms &amp; Conditions
          </a>
        </Link>
        <Link href="/contact">
          <a className="text-black no-underline hover:underline ml-4">
            Contact Us
          </a>
        </Link>
      </div>
    </div>
  </footer>
);

export default memo(Footer);
