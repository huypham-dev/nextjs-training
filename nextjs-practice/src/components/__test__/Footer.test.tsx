import { shallow } from "enzyme";
import Footer from "../common/Footer";

describe("Footer", () => {
  it("Snapshot", () => {
    const component = shallow(<Footer />);

    expect(component).toMatchSnapshot();
  });
});
