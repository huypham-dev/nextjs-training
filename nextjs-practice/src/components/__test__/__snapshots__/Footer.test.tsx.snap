// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot 1`] = `
<footer
  className="w-full bg-white px-6 border-t"
>
  <div
    className="container mx-auto max-w-4xl py-6 flex flex-wrap md:flex-no-wrap justify-between items-center text-sm"
  >
    Â©2021 Your Company. All rights reserved.
    <div
      className="pt-4 md:p-0 text-center md:text-right text-xs"
    >
      <Link
        href="/policy"
      >
        <a
          className="text-black no-underline hover:underline"
        >
          Privacy Policy
        </a>
      </Link>
      <Link
        href="/condition"
      >
        <a
          className="text-black no-underline hover:underline ml-4"
        >
          Terms & Conditions
        </a>
      </Link>
      <Link
        href="/contact"
      >
        <a
          className="text-black no-underline hover:underline ml-4"
        >
          Contact Us
        </a>
      </Link>
    </div>
  </div>
</footer>
`;
